cupy/cuda/stream.pyx

```diff
@@ -214,16 +214,41 @@ class BaseStream(object):
                 object), and returns nothing.
             arg (object): Argument to the callback.
 
-        """
-        if runtime._is_hip_environment and self.ptr == 0:
-            raise RuntimeError('HIP does not allow adding callbacks to the '
-                               'default (null) stream')
+        .. note::
+            Whenever possible, use the :meth:`launch_host_func` method
+            instead of this one, as it may be deprecated and removed from
+            CUDA at some point.
 
+        """
         def f(stream, status, dummy):
             callback(self, status, arg)
 
         runtime.streamAddCallback(self.ptr, f, 0)
 
+    def launch_host_func(self, callback, arg):
+        """Launch a callback on host when all queued work is done.
+
+        Args:
+            callback (function): Callback function. It must take only one
+                argument (user data object), and returns nothing.
```

Comment:
Ah, I am following the requirement from `cudaLaunchHostFunc` that the host function should only have 1 argument instead of 3. 

For the test we would like to do there, we could either use `add_callback` + a 3-arg callback, or use this new function `launch_host_func` + a 1-arg callback, with the stream added as part of the arg:
```python
    data = []
    data.append(stream)
    stream.launch_host_func(callback, data)
```
and check the stream pointer in the callback function.

Reply:
Got it, I missed the fact that's how `cudaLaunchHostFunc` should behave out of my ignorance of that function, thanks for clarifying!

Reply:
No worries, Peter! I didn't know this function before either. I became aware of it only because your PTDS PR made me wonder if it's possible to test it programmatically other than eyeballing `nvvp`/`nsys`, and I started browsing the Runtime API doc 😄